packages/botonic-plugin-flow-builder/src/content-fields/flow-button.tsx

```diff
@@ -1,24 +1,53 @@
 import { Button, Reply } from '@botonic/react'
 import React from 'react'
 
+import { FlowBuilderApi } from '../api'
 import { ContentFieldsBase } from './content-fields-base'
-import { HtButton, HtButtonStyle } from './hubtype-fields'
+import {
+  HtButton,
+  HtButtonStyle,
+  HtPayloadNode,
+  HtUrlNode,
+} from './hubtype-fields'
 
 export class FlowButton extends ContentFieldsBase {
   public text = ''
   public url?: string
   public payload?: string
 
-  static fromHubtypeCMS(component: HtButton, locale: string): FlowButton {
-    const newButton = new FlowButton(component.id)
-    newButton.text = this.getTextByLocale(locale, component.text)
-    newButton.payload = component.target
-      ? component.target.id
-      : component.payload?.find(payload => payload.locale === locale)?.id
-    newButton.url = component.url?.find(url => url.locale === locale)?.id
+  static fromHubtypeCMS(
+    cmsButton: HtButton,
+    locale: string,
+    cmsApi: FlowBuilderApi
+  ): FlowButton {
+    const payloadId = this.getPayloadId(cmsButton, locale)
+    const urlId = this.getUrlId(cmsButton, locale)
+
+    const newButton = new FlowButton(cmsButton.id)
+    newButton.text = this.getTextByLocale(locale, cmsButton.text)
+    if (cmsButton.target) {
+      newButton.payload = cmsButton.target.id
+    }
+    if (cmsButton.payload && payloadId) {
```

Comment:
I think it is a very good idea to do this logic within the content class instead of with functions that are not grouped in any class

Reply:
same here with `new` prefix